simulator/modules/writeback/writeback.cpp

```diff
@@ -55,14 +55,19 @@ auto Writeback<ISA>::read_instructions( Cycle cycle)
 template <typename ISA>
 void Writeback<ISA>::clock( Cycle cycle)
 {
-    sout << "wb      cycle " << std::dec << cycle << ": ";
-
     auto instrs = read_instructions( cycle);
 
     if ( instrs.empty())
+    {
+        sout << "wb      cycle " << std::dec << cycle << ": ";
         writeback_bubble( cycle);
+    }
     else
-        writeback_instruction( instrs.front(), cycle);
+        for( uint32 i = 0; i < bandwidth && i < instrs.size(); i++)
```

Comment:
Please use range-based for:
```c++
for (const auto& i : instrs)
{
    sout << "wb      cycle " << std::dec << cycle << ": ";
    writeback_instruction( i, cycle);
}

Reply:
What would happen if loop counter become greater or equal than bandwidth? Where shall the instructions be?
I think that it might be omitted here.

Reply:
I wrote for loop like that because `instrs.size()` might be 3 if WB gets something from branch, memory and execute modules at the same time. So, if WB_bandwidth is 2 in that case, we have to stall pipeline.